src/main/java/org/sonatype/nexus/blobstore/s3/internal/S3BlobStore.java

```diff
@@ -401,14 +440,51 @@ public void init(final BlobStoreConfiguration configuration) {
       this.s3 = amazonS3Factory.create(configuration);
       if (!s3.doesBucketExist(getConfiguredBucket())) {
         s3.createBucket(getConfiguredBucket());
+
+        addBucketLifecycleConfiguration();
+      } else {
+        // bucket exists, we should test that the correct lifecycle config is present
+        BucketLifecycleConfiguration lifecycleConfiguration = s3.getBucketLifecycleConfiguration(getConfiguredBucket());
+        if (lifecycleConfiguration == null ||
+            !lifecycleConfiguration.getRules().stream()
+                .filter(r -> r.getExpirationInDays() == getConfiguredExpirationInDays())
+                .filter(r -> {
+                  LifecycleFilterPredicate predicate = r.getFilter().getPredicate();
+                  if (predicate instanceof LifecycleTagPredicate) {
+                    LifecycleTagPredicate tagPredicate = (LifecycleTagPredicate) predicate;
+                    return DELETED_TAG.equals(tagPredicate.getTag());
+                  }
+                  return false;
+                })
+                .findAny().isPresent()) {
```

Comment:
This would be a good candidate for a separate method. It would shorten up the `if ` statement, and the method's purpose / readability would be really clear.